### PR TITLE
refs: https://github.com/metasfresh/metasfresh/issues/14651 add unq i…

### PR DESCRIPTION
**Pull request by fp-ps**
…ndex

### CODE REVIEW

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5680260_sys_gh14651_add_C_Project_unq_idx_ExternalID_OrgID_ProjectType.sql [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+INSERT INTO AD_Index_Column (AD_Client_ID,AD_Column_ID,AD_Index_Column_ID,AD_Index_Table_ID,AD_Org_ID,Created,CreatedBy,EntityType,IsActive,SeqNo,Updated,UpdatedBy) VALUES (0,1350,541320,540728,0,TO_TIMESTAMP('2023-03-02 19:37:29','YYYY-MM-DD HH24:MI:SS'),100,'D','Y',20,TO_TIMESTAMP('2023-03-02 19:37:29','YYYY-MM-DD HH24:MI:SS'),100)
+;
+
+-- 2023-03-02T17:37:39.720Z
+INSERT INTO AD_Index_Column (AD_Client_ID,AD_Column_ID,AD_Index_Column_ID,AD_Index_Table_ID,AD_Org_ID,Created,CreatedBy,EntityType,IsActive,SeqNo,Updated,UpdatedBy) VALUES (0,8757,541321,540728,0,TO_TIMESTAMP('2023-03-02 19:37:39','YYYY-MM-DD HH24:MI:SS'),100,'D','Y',30,TO_TIMESTAMP('2023-03-02 19:37:39','YYYY-MM-DD HH24:MI:SS'),100)
+;
+
+-- 2023-03-02T17:37:43.370Z
+CREATE UNIQUE INDEX ExternalID_OrgID_ProjectType_Nullable ON C_Project (ExternalId,AD_Org_ID,C_ProjectType_ID) WHERE C_ProjectType_ID IS NULL and isactive = 'Y'
+;
```
Comment (cp-ps) [inline backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5680260_sys_gh14651_add_C_Project_unq_idx_ExternalID_OrgID_ProjectType.sql:47]: postgres sees `null` values as distinct, so the index must only on C_Project (ExternalId,AD_Org_ID)

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5680260_sys_gh14651_add_C_Project_unq_idx_ExternalID_OrgID_ProjectType.sql [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+-- 2023-03-02T17:36:18.845Z
+INSERT INTO AD_Index_Column (AD_Client_ID,AD_Column_ID,AD_Index_Column_ID,AD_Index_Table_ID,AD_Org_ID,Created,CreatedBy,EntityType,IsActive,SeqNo,Updated,UpdatedBy) VALUES (0,1350,541317,540727,0,TO_TIMESTAMP('2023-03-02 19:36:18','YYYY-MM-DD HH24:MI:SS'),100,'D','Y',20,TO_TIMESTAMP('2023-03-02 19:36:18','YYYY-MM-DD HH24:MI:SS'),100)
+;
+
+-- 2023-03-02T17:36:26.035Z
+INSERT INTO AD_Index_Column (AD_Client_ID,AD_Column_ID,AD_Index_Column_ID,AD_Index_Table_ID,AD_Org_ID,Created,CreatedBy,EntityType,IsActive,SeqNo,Updated,UpdatedBy) VALUES (0,8757,541318,540727,0,TO_TIMESTAMP('2023-03-02 19:36:25','YYYY-MM-DD HH24:MI:SS'),100,'D','Y',30,TO_TIMESTAMP('2023-03-02 19:36:25','YYYY-MM-DD HH24:MI:SS'),100)
+;
+
+-- 2023-03-02T17:36:30.391Z
+CREATE UNIQUE INDEX ExternalID_OrgID_ProjectType_NonNull ON C_Project (ExternalId,AD_Org_ID,C_ProjectType_ID) WHERE C_ProjectType_ID IS NOT NULL and isactive = 'Y'
```
Comment (cp-ps) [inline backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5680260_sys_gh14651_add_C_Project_unq_idx_ExternalID_OrgID_ProjectType.sql:22]: 👍

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5680260_sys_gh14651_add_C_Project_unq_idx_ExternalID_OrgID_ProjectType.sql (hunk 1 above)
Comment (cp-ps) [reply backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5680260_sys_gh14651_add_C_Project_unq_idx_ExternalID_OrgID_ProjectType.sql:47]: maybe also change the name to `ExternalID_OrgID_with_null_project_type`

**Review by cp-ps — CHANGES_REQUESTED** [review]

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5680260_sys_gh14651_add_C_Project_unq_idx_ExternalID_OrgID_ProjectType.sql (hunk 1 above)
Comment (fp-ps) [reply backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5680260_sys_gh14651_add_C_Project_unq_idx_ExternalID_OrgID_ProjectType.sql:57]: index renamed

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5680260_sys_gh14651_add_C_Project_unq_idx_ExternalID_OrgID_ProjectType.sql [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+;
+
+-- 2023-03-03T06:47:00.674Z
+UPDATE AD_Index_Table SET Name='ExternalID_OrgID_with_null_project_type',Updated=TO_TIMESTAMP('2023-03-03 08:47:00','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Index_Table_ID=540728
+;
+
+-- 2023-03-03T06:47:03.079Z
+CREATE UNIQUE INDEX ExternalID_OrgID_with_null_project_type ON C_Project (ExternalId,AD_Org_ID) WHERE C_ProjectType_ID IS NULL and isactive = 'Y'
+;
+
```
Comment (cp-ps) [inline backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5680260_sys_gh14651_add_C_Project_unq_idx_ExternalID_OrgID_ProjectType.sql:79]: at this point we have both indexes: ExternalID_OrgID_ProjectType_Nullable && ExternalID_OrgID_with_null_project_type, pls drop ExternalID_OrgID_ProjectType_Nullable

**Review by cp-ps — CHANGES_REQUESTED** [review]

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5680260_sys_gh14651_add_C_Project_unq_idx_ExternalID_OrgID_ProjectType.sql (hunk 1 above)
Comment (metas-ts) [reply backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5680260_sys_gh14651_add_C_Project_unq_idx_ExternalID_OrgID_ProjectType.sql:57]: @fp-ps @cp-ps you can use `ExternalId,AD_Org_ID,coalesce(C_ProjectType_ID,0)` 
so maybe not two indixes are needed after all?

**Review by cp-ps — APPROVED** [review]

**Pull request merged**